### PR TITLE
Remove experimental_when_header feature

### PR DESCRIPTION
**Pull request by bnjjj**
If you used the `experimental_when_header` feature previously like this for example:

```yaml title="router.previous.yaml"
telemetry:
  exporters:
    logging:
      # If one of these headers matches we will log supergraph and subgraphs requests/responses
      experimental_when_header: # REMOVED
        - name: apollo-router-log-request
          value: my_client
          headers: true # default: false
          body: true # default: false
```

This feature no longer exists and can be replaced with another one included in custom telemetry. Here is how you would configure custom telemetry to achieve the same result:

```yaml title="router.yaml"
telemetry:
  instrumentation:
    events:
      router:
        request: # Display router request log
          level: info
          condition:
            eq:
            - request_header: apollo-router-log-request
            - my_client
        response: # Display router response log
          level: info
          condition:
            eq:
            - request_header: apollo-router-log-request
            - my_client
      supergraph:
        request: # Display supergraph request log
          level: info
          condition:
            eq:
            - request_header: apollo-router-log-request
            - my_client
        response:
          level: info
          condition:
            eq:
            - request_header: apollo-router-log-request
            - my_client
      subgraph:
        request: # Display subgraph request log
          level: info
          condition:
            eq:
            - supergraph_request_header: apollo-router-log-request
            - my_client
        response: # Display subgraph response log
          level: info
          condition:
            eq:
            - supergraph_request_header: apollo-router-log-request
            - my_client
```

[ROUTER-786]: https://apollographql.atlassian.net/browse/ROUTER-786?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by abernix** [comment]
@bnjjj @garypen Does this get a place in some section of the migration guide?

**Comment by bnjjj** [comment]
@abernix Yes. Do we have an existing one somewhere ?

**Comment by abernix** [comment]
@bnjjj Yup! https://github.com/apollographql/router/blob/next/docs/source/reference/migration/from-router-v1.mdx

File: apollo-router/src/plugins/telemetry/config_new/logging.rs [hunk 1]
```diff
@@ -35,43 +33,11 @@ pub(crate) struct Logging {
...
     /// Settings for logging to a file.
     pub(crate) file: File,
-
-    /// Log configuration to log request and response for subgraphs and supergraph
-    /// Note that this will be removed when events are implemented.
-    #[serde(rename = "experimental_when_header")]
-    pub(crate) when_header: Vec<HeaderLoggingCondition>,
 }
 
 impl Logging {
```
Comment (garypen) [inline apollo-router/src/plugins/telemetry/config_new/logging.rs:38]: I think we could remove this completely.

File: apollo-router/tests/integration/lifecycle.rs [hunk 2]
```diff
@@ -225,30 +225,6 @@ async fn test_cli_config_preview() {
     );
 }
 
-#[tokio::test(flavor = "multi_thread")]
```
Comment (garypen) [inline apollo-router/tests/integration/lifecycle.rs:228]: Is this a test we should retain but use a different experimental feature for testing?

**Review by garypen — CHANGES_REQUESTED** [review]
Needs an entry in the migration guide.

File: docs/source/reference/router/telemetry/instrumentation/selectors.mdx [hunk 3]
```diff
@@ -89,6 +89,7 @@ The subgraph service executes multiple times during query execution, with each e
 | `supergraph_operation_kind` | Yes         | `string`         | The operation kind from the supergraph query                                   |
 | `supergraph_query`          | Yes         | `string`         | The graphql query to the supergraph                                            |
 | `supergraph_query_variable` | Yes         |                  | The name of a supergraph query variable                                        |
+| `supergraph_request_header` | Yes         |                  | The name of a supergraph request header                                        |
```
[the remark this answers is not in the document]
Comment (bnjjj) [reply docs/source/reference/router/telemetry/instrumentation/selectors.mdx:92]: I'll add it on `dev` too

File: docs/source/reference/migration/from-router-v1.mdx [hunk 4]
```diff
@@ -58,6 +58,69 @@ telemetry:
...
+    logging:
+      # If one of these headers matches we will log supergraph and subgraphs requests/responses
+      experimental_when_header: # REMOVED
+        - name: apollo-router-log-request
+          value: my_client
+          headers: true # default: false
+          body: true # default: false
+```
+
+This feature not longer exists and can be replaced with another one included in custom telemetry. Here is an example on how you could have the same behavior with the new configuration:
```
Comment (garypen) [inline docs/source/reference/migration/from-router-v1.mdx:77]:
```suggestion
This feature no longer exists and can be replaced with another one included in custom telemetry. Here is how you would configure custom telemetry to achieve the same result:
```

**Review by garypen — APPROVED** [review]

File: docs/source/reference/router/telemetry/instrumentation/selectors.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (bnjjj) [reply docs/source/reference/router/telemetry/instrumentation/selectors.mdx:92]: https://github.com/apollographql/router/pull/6327

**Pull request merged**